test/validation/api/pktio/pktio.c

```diff
@@ -33,7 +33,10 @@
 #define PKTIO_TS_INTERVAL      (50 * ODP_TIME_MSEC_IN_NS)
 #define PKTIO_TS_MIN_RES       1000
 #define PKTIO_TS_MAX_RES       10000000000
-#define PKTIO_TS_CMP_RES       1
+
+/* Increase arithmetic tolerance to accommodate non-integer HW frequency
+   values */
+#define PKTIO_TS_CMP_RES       2
```

Comment:
Actually, the resolution API tells in which granules packet IO timestamp is able to measure time. So, instead of +-2nsec tolerance, the correction could be +-res nsec tolerance. E.g. a 100MHz clock resolution could round 105nsec to a 100 or 110nsec timestamp.

The time validation test uses also resolution from API in tolerance calculations.

``
	/* Test odp_pktio_ts_res() and odp_pktio_ts_from_ns() */
	res = odp_pktio_ts_res(pktio_tx);
	CU_ASSERT(res > PKTIO_TS_MIN_RES);
	CU_ASSERT(res < PKTIO_TS_MAX_RES);
	ns1 = 100;
	ts = odp_pktio_ts_from_ns(pktio_tx, ns1);
	ns2 = odp_time_to_ns(ts);
	/* Allow some arithmetic tolerance */
	CU_ASSERT((ns2 <= (ns1 + PKTIO_TS_CMP_RES)) &&
		  (ns2 >= (ns1 - PKTIO_TS_CMP_RES)));
``